Cherry-pick VS insertion automation to release/17.4

allisonchou opened the pull request:
﻿### Summary of the changes

- We need to insert into 17.4 (see https://github.com/dotnet/razor/pull/8097). However, as part of https://github.com/dotnet/razor/pull/7944 to automate VS insertions, I updated the drop folder to point to a different location. We should cherry-pick this change (+ related changes) back to 17.4 so we can insert.

cc @Cosifne

davidwengier commented:
> updated the drop folder to point to a different location.

And this doesn't happen in a .yml file?

allisonchou commented:
> And this doesn't happen in a .yml file?

I updated the yml file so the drop path is now `DropFolder: 'artifacts\VSSetup\$(_BuildConfig)\Insertion'`, and updated the drop path in the [release pipeline](https://dev.azure.com/dnceng/internal/_release?definitionId=18&view=mine&_a=releases) to match. Hypothetically, we could temporarily change back the release pipeline to the old path, but we're going to run into the same issue if we service 17.4 again, so I thought it'd be more efficient in the long run to cherry-pick this change.

davidwengier commented:
All good, just making sure I understand, because the PR you linked to that caused the break was after 17.4 :)

allisonchou commented:
Also @davidwengier - messaged you separately about this just now too, but I'm not sure if cherry-picking this will cause problems in VS since we're once again changing the version #?
